armi/physics/neutronics/crossSectionGroupManager.py

```diff
@@ -1002,46 +1015,85 @@ def _setBuGroupBounds(self, upperBuGroupBounds):
-            for block in blockList:
-                bu = block.p.percentBu
-                for buGroupIndex, upperBu in enumerate(self._upperBuGroupBounds):
-                    if bu <= upperBu:
-                        block.p.buGroupNum = buGroupIndex
-                        break
-                else:
-                    raise ValueError("no bu group found for bu={0}".format(bu))
-        else:
+        if not self._envGroupUpdatesEnabled:
             runLog.debug(
                 "Skipping burnup group update of {0} blocks because it is disabled"
                 "".format(len(blockList))
             )
+            return
+
+        numBuGroups = len(self._buGroupBounds)
+        runLog.debug("Updating env groups of {0} blocks".format(len(blockList)))
+        if len(self._buGroupBounds) == 1 and len(self._tempGroupBounds) == 1:
+            # dont set block.p.envGroupNum since all 1 group and we want to support 2 char xsGroup
+            return
+        for block in blockList:
+            bu = block.p.percentBu
+            for buIndex, upperBu in enumerate(self._buGroupBounds):
+                if bu <= upperBu:
+                    buGroupVal = buIndex
+                    tempGroupVal = 0
+                    isotope = self._initializeXsID(block.getMicroSuffix()).xsTempIsotope
+                    if isotope and len(self._tempGroupBounds) > 1:
+                        # if statement saves this somewhat expensive calc if we are not doing temp groups
+                        tempC = getBlockNuclideTemperature(block, isotope)
+                        for tempIndex, upperTemp in enumerate(self._tempGroupBounds):
+                            if tempC <= upperTemp:
+                                tempGroupVal = tempIndex
+                                break
+                    # this ordering groups like-temperatures together in group number
+                    block.p.envGroupNum = tempGroupVal * numBuGroups + buGroupVal
+                    break
 
     def _addXsGroupsFromBlocks(self, blockCollectionsByXsGroup, blockList):
         """
         Build all the cross section groups based on their XS type and BU group.
 
         Also ensures that their BU group is up to date with their burnup.
         """
-        self._updateBurnupGroups(blockList)
+        self._updateEnvironmentGroups(blockList)
         for b in blockList:
             xsID = b.getMicroSuffix()
             xsSettings = self._initializeXsID(xsID)
+            if (
+                self.cs["tempGroups"]
+                and xsSettings.blockRepresentation == MEDIAN_BLOCK_COLLECTION
+            ):
+                runLog.warning(
+                    "Median block currently only consider median burnup block, and "
+                    "not median temperature block in group"
+                )
             blockCollectionType = blockCollectionFactory(
                 xsSettings, self.r.blueprints.allNuclidesInProblem
             )
+
```

Comment:
We might need to make a call to `calcAvgNuclideTemperatures()` right here in order for the temperatures to be available for grouping.

```suggestion
+        for xsID, blockCollection in blockCollectionsByXsGroup.items():
+            blockCollection.calcAvgNuclideTemperatures()
```

I think this is right but I'll need to verify with in-use testing.

Reply:
i added to cylindrical instead
 
![image](https://github.com/user-attachments/assets/12558265-2ca9-42ef-96c1-85517a911999)